native/dpotf2.go

```diff
@@ -0,0 +1,66 @@
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dpotrf computes the cholesky decomposition of the symmetric positive definite
+// matrix a. If ul == blas.Upper, then a is stored as an upper-triangular matrix,
+// and a = U^T U is stored in place into a. If ul == blas.Lower, then a = L L^T
+// is computed and stored in-place into a. This is the unblocked version of the
+// algorithm. Dpotrf returns 0 if the decomposition is successful, otherwise
```

Comment:
Maybe? If I had a matrix I thought was positive definite, and Chol told me it wasn’t, my next step would be to look at the eigenvalues to see if they’re all positive. If they aren’t and I wanted to dig further, I’m not sure how to do that other than to intrusively modify the go cholesky routine to do some print statements. I’m not sure what I’d learn from the number itself.

> On May 7, 2015, at 5:06 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In native/dpotf2.go <https://github.com/gonum/lapack/pull/11#discussion_r29907823>:
> 
> > @@ -0,0 +1,66 @@
> > +package native
> > +
> > +import (
> > +	"math"
> > +
> > +	"github.com/gonum/blas"
> > +	"github.com/gonum/blas/blas64"
> > +)
> > +
> > +// Dpotrf computes the cholesky decomposition of the symmetric positive definite
> > +// matrix a. If ul == blas.Upper, then a is stored as an upper-triangular matrix,
> > +// and a = U^T U is stored in place into a. If ul == blas.Lower, then a = L L^T
> > +// is computed and stored in-place into a. This is the unblocked version of the
> > +// algorithm. Dpotrf returns 0 if the decomposition is successful, otherwise
> I guess I'm wondering if it would be useful for debugging.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/gonum/lapack/pull/11/files#r29907823>.
>